dissect/util/feature.py

```diff
@@ -0,0 +1,54 @@
+import functools
+import os
+from enum import Enum
+from typing import Callable, Optional
+
+
+# Register feature flags in a central place to avoid chaos
+class Feature(Enum):
+    ADVANCED = "advanced"
+    LATEST = "latest"
+    BETA = "beta"
+
+
+# Defines the default flags (as strings)
+DISSECT_FEATURES_DEFAULT = "latest"
+
+# Defines the environment variable to read the flags from
+DISSECT_FEATURES_ENV = "DISSECT_FEATURES"
+
+
+class FeatureException(RuntimeError):
+    pass
+
+
+@functools.cache
+def feature_flags() -> list[Feature]:
+    return [Feature(name) for name in os.getenv(DISSECT_FEATURES_ENV, DISSECT_FEATURES_DEFAULT).split("/")]
+
+
+@functools.cache
+def feature_enabled(feature: Feature) -> bool:
```

Comment:
Can you add a docstring that includes a code example (if statement) that uses this?